src/rate_limiter.rs

```diff
@@ -151,10 +151,11 @@ impl RateLimiter {
             .map(|(ip_addr, used)| (*used, *ip_addr))
             .collect();
         entries.sort();
+        self.used.clear();
         for (index, (used, client)) in entries.into_iter().enumerate() {
             if used < quota {
                 leaked_units -= used;
-                quota = leaked_units / (leaking_client_count - index + 1) as u64;
+                quota = leaked_units / (leaking_client_count - index) as u64;
```

Comment:
I'm not sure this change is correct.  Say we have `leaked_units == 500` and `leaking_client_count == 5` with the first client only having used 20.  Initially their `qutoa` is 100 each.  In the first iteration of this for loop, we don't want to re-insert the client back into `self.used`, but we do need to recalculate the quota for the other four since this one didn't use his full 100.  The recalculation should yield `480/4` as the new quota, not `480/5`.